facebookresearch/fmmax
# Initial docs push

Here we add official docs for the package. These docs will be hosted [here](https://facebookresearch.github.io/fmmax/). In particular, this PR adds the following functionality:

* A formal docs site powered via docusaurus (preferred by Meta).
* Automatic generation of an API from the abstract source tree itself. Does not require importing the package.
* A set of automatic deployment tools (via GitHub actions) which will trigger when a PR is created and merged. The former checks if the build is successful, the latter deploys the exported site to the appropriate branch.

To test locally:
1. Ensure you have [Node.js](https://nodejs.org/en/download/) installed
2. `$ cd docs`
3. `$ npx docusaurus start`

To automatically generate a new API:
1. `$ pip install docstring-parser`
2. `$ cd docs`
3. `$ python gen_API_from_docstrings.py`
